### ISSUE TITLE
AttributeError: 'LanguageModel' object has no attribute 'document_delimiter'

### ISSUE THREAD

**Issue by sgraaf** [flairNLP/flair#2503]
**Describe the bug**
Hi! I am trying to "fine-tune" an existing, pre-trained `SequenceTagger` model (specifically `flair/ner-english-ontonotes-fast`) on a proprietary, custom `ColumnCorpus` with completely different entity labels. Inspired by a comment in #1540, for this I manipulate the pre-trained model's `state`, after which I run the training using the custom `Corpus`. 

After training for `X` epochs (I set `X` to 10), when the trainer saves the `final-model.pt`, I am faced with the following error:
```
[...]
2021-11-04 16:10:35,817 EPOCH 10 done: loss 0.2134 - lr 0.0100000
2021-11-04 16:10:36,329 DEV : loss 0.2844020426273346 - f1-score (micro avg)  0.0
2021-11-04 16:10:36,338 BAD EPOCHS (no improvement): 0
Traceback (most recent call last):
  File "C:\Users\sgraaf\Documents\GitHub\NER-Project\src\fine-tune.py", line 116, in <module>
    main()
  File "C:\Users\sgraaf\Documents\GitHub\NER-Project\src\fine-tune.py", line 107, in main
    trainer.train(
  File "C:\Users\sgraaf\Documents\GitHub\NER-Project\.venv\lib\site-packages\flair\trainers\trainer.py", line 690, in train
    self.model.save(base_path / "final-model.pt")
  File "C:\Users\sgraaf\Documents\GitHub\NER-Project\.venv\lib\site-packages\flair\nn\model.py", line 85, in save
    torch.save(model_state, str(model_file), pickle_protocol=4)
  File "C:\Users\sgraaf\Documents\GitHub\NER-Project\.venv\lib\site-packages\torch\serialization.py", line 379, in save
    _save(obj, opened_zipfile, pickle_module, pickle_protocol)
  File "C:\Users\sgraaf\Documents\GitHub\NER-Project\.venv\lib\site-packages\torch\serialization.py", line 484, in _save
    pickler.dump(obj)
  File "C:\Users\sgraaf\Documents\GitHub\NER-Project\.venv\lib\site-packages\flair\models\language_model.py", line 414, in __getstate__
    "document_delimiter": self.document_delimiter,
  File "C:\Users\sgraaf\Documents\GitHub\NER-Project\.venv\lib\site-packages\torch\nn\modules\module.py", line 1130, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(
AttributeError: 'LanguageModel' object has no attribute 'document_delimiter'
```

**To Reproduce**
Using the following script:...
```Python
from argparse import ArgumentParser
from pathlib import Path

import torch
from flair.datasets import ColumnCorpus
from flair.models import SequenceTagger
from flair.models.sequence_tagger_model import START_TAG, STOP_TAG
from flair.trainers import ModelTrainer

TAG_TYPE = "ner"


def main():
    parser = ArgumentParser(description="Parse annotation data")
    parser.add_argument(
        "-i",
        "--input_file",
        type=Path,
        required=True,
        help="Path to the input file containing the (raw) annotations.",
        metavar="PATH",
    )
    parser.add_argument(
        "-m",
        "--model",
        type=str,
        required=True,
        help="The `flair` model to fine-tune on the input file.",
        metavar="MODEL",
    )
    parser.add_argument(
        "-o",
        "--output_dir",
        type=Path,
        required=True,
        help="The output directory where all output (logs, weights, etc.) will be saved.",
        metavar="PATH",
    )

    args = parser.parse_args()

    # create the output directory
    args.output_dir.mkdir(parents=True, exist_ok=True)

    # load the corpus
    corpus = ColumnCorpus(
        data_folder=args.input_file.parent,
        column_format={0: "text", 1: "ner"},
        train_file=args.input_file.name,
        tag_to_bioes="ner",
    )

    # make the tag dictionary from the corpus
    tag_dictionary = corpus.make_tag_dictionary(tag_type=TAG_TYPE)
    tagset_size = len(tag_dictionary)

    # load the tagger model to fine-tune
    tagger_ = SequenceTagger.load(args.model)

    # manipulate the tagger's `state`
    # see: https://github.com/flairNLP/flair/issues/1540#issuecomment-619287834
    state = tagger_._get_state_dict()

    # update `tag_dictionary`
    state["tag_dictionary"] = tag_dictionary

    # randomly initialize transitions
    state["state_dict"]["transitions"] = torch.nn.Parameter(
        torch.randn(tagset_size, tagset_size)
    )
    state["state_dict"]["transitions"].detach()[
        tag_dictionary.get_idx_for_item(START_TAG), :
    ] = -10000
    state["state_dict"]["transitions"].detach()[
        :, tag_dictionary.get_idx_for_item(STOP_TAG)
    ] = -10000

    # randomly initialize linear layer
    num_directions = 2 if tagger_.bidirectional else 1
    linear_layer = torch.nn.Linear(
        tagger_.hidden_size * num_directions, len(tag_dictionary)
    )
    state["state_dict"]["linear.weight"] = linear_layer.weight
    state["state_dict"]["linear.bias"] = linear_layer.bias

    # re-initialize the tagger with the manipulated `state`
    tagger = SequenceTagger._init_model_with_state_dict(state)

    # initialize the trainer
    trainer = ModelTrainer(tagger, corpus)

    # run training
    trainer.train(
        base_path=args.output_dir,
        learning_rate=0.01,
        mini_batch_size=32,
        max_epochs=10,
    )


if __name__ == "__main__":
    main()
```

... with the following arguments:...
```
--input_file path/to/custom/iob_formatted_file.txt \
--model "flair/ner-english-ontonotes-fast" \
--output_dir path/to/output/directory
```

.. procudes the above error.

**Expected behavior**
Simply put, I did not expect any errors.

**Environment (please complete the following information):**
 - OS [e.g. iOS, Linux]: Windows 10, build 19043.1288
 - Version [e.g. flair-0.3.2]: `flair-0.9`

**Comment by Tarotis** [comment]
I reproduce this error with the shorter code:
```
from flair.datasets import UD_ENGLISH
from flair.models import SequenceTagger
from flair.trainers import ModelTrainer

corpus = UD_ENGLISH().downsample(0.005)
tagger = SequenceTagger.load("pos-multi-fast")
trainer = ModelTrainer(tagger, corpus)
trainer.fine_tune('resources/taggers/example-upos', 
                    learning_rate=0.1, 
                    mini_batch_size=8, 
                    max_epochs=2, 
                    #param_selection_mode=True,
                    )
```

I was able to fine-tune the model by setting `param_selection_mode` to `True`, but then only the last model gets saved, so the problem continues.
